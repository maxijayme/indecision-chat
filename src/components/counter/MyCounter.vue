<script lang="ts" setup>
import { useCounter } from '../../composables/useCounter';

interface Props {
  value: number;
}
const props = defineProps<Props>();

const { counter, squareCounter } = useCounter(props.value);

// import { computed, ref } from 'vue';

// const counter = ref(props.value);
// const squareCounter = computed(() => counter.value * counter.value);
</script>

<template>
  <section>
    <h3 data-vtest="counter-label">Counter: {{ counter }}</h3>
    <h3>Square: {{ squareCounter }}</h3>

    <div>
      <button data-vtest="increment-btn" class="btn" @click="counter++">+1</button>
      <button data-vtest="decrement-btn" class="btn" @click="counter--">-1</button>
    </div>
  </section>
</template>

<style scoped>

</style>
