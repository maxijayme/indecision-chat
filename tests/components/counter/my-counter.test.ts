import { mount } from '@vue/test-utils'
import MyCounter from '@/components/counter/MyCounter.vue'

describe('<MyCounter>', () => {
  const counterValue = 10
  test('should match snapshot', () => {
    const wrapper = mount(MyCounter, {
      props: {
        value: counterValue,
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('should render counter label', () => {
    const wrapper = mount(MyCounter, {
      props: {
        value: counterValue,
      },
    })
    const counterLabel = wrapper.get('[data-vtest="counter-label"]')
    expect(counterLabel.text()).toBe(`Counter: ${counterValue}`)
  })

  test('should render both h3 lables', () => {
    const wrapper = mount(MyCounter, {
      props: {
        value: counterValue,
      },
    })
    const [ counterLable, SquareLabel ] = wrapper.findAll('h3')
    expect(counterLable.text()).toBe(`Counter: ${counterValue}`)
    expect(SquareLabel.text()).toBe(`Square: ${counterValue * counterValue}`)
  })

  test('should increment counter + 1 on click', async () => {
    const wrapper = mount(MyCounter, {
      props: {
        value: counterValue,
      },
    })

    const incrementButton = wrapper.get('[data-vtest="increment-btn"]')
    const counterLabel = wrapper.get('[data-vtest="counter-label"]')
    await incrementButton.trigger('click')
    expect(counterLabel.text()).toContain(`Counter: ${counterValue + 1}`)
  })
})
