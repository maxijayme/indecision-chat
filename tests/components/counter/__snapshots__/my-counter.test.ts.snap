// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MyCounter> > should match snapshot 1`] = `
<section>
  <h3
    data-vtest="counter-label"
  >
    Counter: 10
  </h3>
  <h3>
    Square: 100
  </h3>
  <div>
    <button
      class="btn"
      data-vtest="increment-btn"
    >
      +1
    </button>
    <button
      class="btn"
      data-vtest="decrement-btn"
    >
      -1
    </button>
  </div>
</section>
`;
